src/Schema/SqliteSchemaManager.php

```diff
@@ -443,7 +443,7 @@ protected function _getPortableTableForeignKeysList($tableForeignKeys)
         $list = [];
         foreach ($tableForeignKeys as $value) {
             $value = array_change_key_case($value, CASE_LOWER);
-            $name  = $value['constraint_name'];
+            $name  = $value['constraint_name'] !== '' ? $value['constraint_name'] : count($list);
```

Comment:
Please consider:

```sql
CREATE TABLE artist
(
    id   INTEGER,
    name VARCHAR(32)
);

CREATE TABLE track
(
    id        INTEGER,
    name      VARCHAR(32),
    artist_id INTEGER,
    CONSTRAINT "0" FOREIGN KEY (artist_id) REFERENCES artist(id)
);
```

No matter what name is implied for an unnamed key, it can always conflict with another named one.

Reply:
Note that as of 4.0.0 (e.g. https://github.com/doctrine/dbal/pull/4777), table columns, schema tables, and database schemas are represented as lists rather than maps of object names to their definitions.

We should consider doing the same for all types of objects.

Reply:
Very good point 👍 - I tested `CONSTRAINT 0...` but not `CONSTRAINT "0"`.

I have now prefixed the index name with some "highly unlikely string". And yes, in 4.0.x, indexing as list like in #4777 is welcomed.

Reply:
That's a hack. We need a proper fix instead.

Reply:
Yes, it is, but it will be otherwise BC break, so for 3.x, I do not see any other solution.

Reply:
We need to implement a proper solution first and then, if possible, backport it or define the upgrade path. I don't think hacking the 3.x code just for the sake of addressing an edge case is a good idea.